packages/angular/projects/ui-angular/src/lib/primitives-poc/text-field/text-field-component/text-field.component.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  AfterContentInit,
+  Component,
+  ContentChild,
+  Input,
+  SimpleChanges,
+} from '@angular/core';
+import { InputDirective } from '../text-field-directives/input.directive';
+
+const $COMPONENT_SELECTOR: string = 'amplify-poc-text-field';
+@Component({
+  selector: $COMPONENT_SELECTOR,
+  templateUrl: 'text-field.component.html',
+})
+export class AmplifyTextFieldsComponent implements AfterContentInit {
+  @Input() describedBy: string;
+  @Input() labelHidden: boolean = false;
+  @Input() hasError: boolean = false;
+
+  @ContentChild(InputDirective) childInput: InputDirective | undefined;
+
+  public inputId = '';
+
+  constructor() {}
+
+  ngAfterContentInit(): void {
+    // getting data from child input
+    this.inputId = this.childInput?.id;
+
+    // send data to child input
+    // note: this is still not dynamic, so we still need a change detction strategy here.
+    // this can be implemented later for the purpose of POC.
+    this.childInput.hasError = this.hasError;
+    this.childInput.ariaDescribedBy = this.describedBy;
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
```

Comment:
Nice! In the future, we'll need to find a mechanism to better maintain this, but this LGTM for now.